keps/sig-node/20190221-maxInitialFailureCount-health-probes.md

```diff
@@ -0,0 +1,145 @@
+---
+title: Add maxInitialFailureCount to health probes
+authors:
+  - "@matthyx"
+owning-sig: sig-node
+participating-sigs:
+  - sig-apps
+  - sig-architecture
+reviewers:
+  - @RobertKrawitz
+approvers:
+  - @RobertKrawitz
+editor: TBD
+creation-date: 2019-02-21
+last-updated: 2019-02-28
+status: provisional
+see-also:
+replaces:
+superseded-by:
+---
+
+# Add maxInitialFailureCount to health probes
+
+## Table of Contents
+
+- [Add maxInitialFailureCount to health probes](#add-maxinitialfailurecount-to-health-probes)
+  - [Table of Contents](#table-of-contents)
+  - [Release Signoff Checklist](#release-signoff-checklist)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [Implementation Details](#implementation-details)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Design Details](#design-details)
+    - [Test Plan](#test-plan)
+  - [Implementation History](#implementation-history)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Release Signoff Checklist
+
+- [ ] kubernetes/enhancements issue in release milestone, which links to KEP (this should be a link to the KEP location in kubernetes/enhancements, not the initial KEP PR)
+- [ ] KEP approvers have set the KEP status to `implementable`
+- [ ] Design details are appropriately documented
+- [ ] Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [ ] Graduation criteria is in place
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/website]: https://github.com/kubernetes/website
+
+## Summary
+
+Slow starting containers are difficult to address with the current status of health probes: they are either killed before being up, or could be left dead-locked during a very long time before being killed.
+
+This proposal adds a numerical option `maxInitialFailureCount` to probes allowing a greater number of failures during the initial start of the container before taking action, while keeping `failureThreshold` at a minimum to restart dead-locked containers after an acceptable delay.
+
+## Motivation
+
+The main problem with slow starting containers is that you have to leave them enough time to start before having `livenessProbe` fail `failureThreshold` times (otherwise `kubelet` kills them before they have a chance to be up).
```

Comment:
Nit: again, don't write in the second person.

More substantively, I'm not quite sure what the "problem" is here from the way it's written.  I can see a few problems, but you need to be explicit about what you're trying to solve here.